rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport.hpp

```diff
@@ -56,11 +58,15 @@ struct StringHelper<rosidl_typesupport_introspection_c__MessageMembers>
   {
     auto c_string = static_cast<rosidl_generator_c__String *>(data);
     if (!c_string) {
-      fprintf(stderr, "Failed to cast data as rosidl_generator_c__String\n");
+      RCUTILS_LOG_ERROR_NAMED(
+        "rmw_fastrtps_cpp",
+        "Failed to cast data as rosidl_generator_c__String")
```

Comment:
referenced this in the [audit rmw_fastrtps](https://github.com/ros2/rmw_fastrtps/issues/122) issue